sunpy/time/time.py

```diff
@@ -156,18 +164,20 @@ def parse_time(time_string, time_format='', **kwargs):
     """
     if isinstance(time_string, pandas.Timestamp):
         return time_string.to_pydatetime()
+    elif isinstance(time_string, pandas.Series) and 'datetime64' in str(time_string.dtype):
+        return np.array([dt.to_pydatetime() for dt in time_string])
+    elif isinstance(time_string, pandas.DatetimeIndex):
+        return time_string._mpl_repr()
```

Comment:
`_mpl_repr`? what's that an array of datetimes? Knowing that pandas loves to be a moving target I would much rather this not be a private method in pandas land.

Reply:
No idea, I just moved that `elif` block up from below to sit next to the other pandas blocks.